cmd/init.go

```diff
@@ -0,0 +1,556 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/amp-labs/cli/flags"
+	"github.com/amp-labs/cli/logger"
+	"github.com/amp-labs/cli/openapi"
+	"github.com/amp-labs/cli/request"
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/yaml"
+)
+
+var ErrBadInput = errors.New("bad input")
+
+const (
+	NumProvidersShown = 10
+	YamlFileMode      = 0o644
+)
+
+func nonEmpty(fieldName string) func(string) error {
+	return func(s string) error {
+		if s == "" {
+			return fmt.Errorf("%w: %s cannot be empty", ErrBadInput, fieldName)
+		}
+
+		return nil
+	}
+}
+
+var initCmd = &cobra.Command{ //nolint:gochecknoglobals
+	Use:   "init",
+	Short: "create a new Ampersand project",
+	Long:  "Create a new Ampersand project.",
+	Run: func(cmd *cobra.Command, args []string) {
+		name, err := promptString("Name your integration", nonEmpty("Integration name"))
+		if err != nil {
+			logger.FatalErr("Unable to prompt for integration name", err)
+		}
+
+		provider, err := selectProvider(cmd.Context())
+		if err != nil {
+			logger.FatalErr("Unable to select provider", err)
+		}
+
+		disp := "my integration"
+
+		var integ openapi.Integration
+		integ.Name = name
+		integ.DisplayName = &disp
+		integ.Provider = provider.Name
+
+		if provider.Support.Proxy {
+			if err := setupProxy(&integ, provider); err != nil {
+				logger.FatalErr("Unable to setup proxy", err)
+			}
+		}
+
+		if provider.Support.Read {
+			if err := setupRead(&integ, provider); err != nil {
+				logger.FatalErr("Unable to setup read", err)
+			}
+		}
+
+		if provider.Support.Write {
+			if err := setupWrite(&integ, provider); err != nil {
+				logger.FatalErr("Unable to setup write", err)
+			}
+		}
+
+		path, err := promptString("Save YAML to file path", nonEmpty("File path"))
+		if err != nil {
+			logger.FatalErr("Unable to prompt for file path", err)
+		}
+
+		manifest := &openapi.Manifest{
+			SpecVersion: "1.0.0",
```

Comment:
can you add a TODO that we should add an API endpoint to get the latest spec version from the server, if if the existing CLI that the user has downloaded does not support the latest, then it should block the user from using this init command until they upgrade, that way we don't have a bunch of outdated CLIs in the wild creating outdated manifest files.